### PR TITLE
BUG: MaskedArray.count treats negative axes incorrectly

### LINKED ISSUE #7509: a possible MA regression
scipy tests started failing today with numpy from travis-dev-wheels. 
The failure is in mstats: https://travis-ci.org/scipy/scipy/builds/120882377

The failing test does this:

```
In [1]: from scipy.stats import mstats

In [2]: %paste
        x = [1, 3, 5, 7, 9]
        y = [2, 4, 6, 8, 10]
## -- End pasted text --

In [3]: mstats.kruskal(x, y)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-5edb46bc91da> in <module>()
----> 1 mstats.kruskal(x, y)

/home/br/virtualenvs/np/local/lib/python2.7/site-packages/scipy/stats/mstats_basic.pyc in kruskal(*args)
   1047     ngrp = ranks.count(-1)
   1048     ntot = ranks.count()
-> 1049     H = 12./(ntot*(ntot+1)) * (sumrk**2/ngrp).sum() - 3*(ntot+1)
   1050     # Tie correction
   1051     ties = count_tied_groups(ranks)

/home/br/repos/numpy/build/testenv/lib/python2.7/site-packages/numpy/ma/core.pyc in __truediv__(self, other)
   3983         if self._delegate_binop(other):
   3984             return NotImplemented
-> 3985         return true_divide(self, other)
   3986 
   3987     def __rtruediv__(self, other):

/home/br/repos/numpy/build/testenv/lib/python2.7/site-packages/numpy/ma/core.pyc in __call__(self, a, b, *args, **kwargs)
   1126         # Get the result
   1127         with np.errstate(divide='ignore', invalid='ignore'):
-> 1128             result = self.f(da, db, *args, **kwargs)
   1129         # Get the mask as a combination of the source masks and invalid
   1130         m = ~umath.isfinite(result)

ValueError: operands could not be broadcast together with shapes (2,) (2,5) 
```

and bisecting indicates

```
$ git bisect bad
36f76ea2e6e91062df12d3a46ccaed7822bc82f2 is the first bad commit
commit 36f76ea2e6e91062df12d3a46ccaed7822bc82f2
Author: Allan Haldane <allan.haldane@gmail.com>
Date:   Sun Feb 14 02:49:19 2016 -0500

    ENH: add extra kwargs and update doc of many MA methods
    
    Updated any, all, sum, prod, cumsum, cumprod, min, max, argmin, argmax,
    mean, var

:040000 040000 e41cfd58bb4bbabc11471a0bd443b24be91b45d0 7f99716492632508bc4d664d424c7cf22d6cd5cb M	numpy
```

which is, it seems, gh-5706?

### PR DESCRIPTION
**Pull request by ahaldane**
Follow up to #5706. Fixes #7509

### CODE REVIEW

File: numpy/ma/core.py [hunk 1]
```diff
@@ -4319,6 +4319,11 @@ def count(self, axis=None, keepdims=np._NoValue):
                 return self.size
 
             axes = axis if isinstance(axis, tuple) else (axis,)
+            axes = tuple(a if a >= 0 else self.ndim + a for a in axes)
+            if len(axes) != len(set(axes)):
+                raise ValueError("duplicate value in 'axis'")
+            if np.any([a < 0 or a >= self.ndim for a in axes]):
+                raise ValueError("'axis' entry is out of bounds")
```
Comment (charris) [inline numpy/ma/core.py]: Maybe `IndexError`?
> exception IndexError
>
>    Raised when a sequence subscript is out of range. (Slice indices are silently truncated to fall in the allowed range; if an index is not an integer, TypeError is raised.)

Comment (ahaldane) [inline numpy/ma/core.py]: I'm just copying ufunc behavior here:
```
>>> np.ones((2,3)).all(axis=8)
ValueError: 'axis' entry is out of bounds
```

**Comment by charris** [comment]
OK, we'll go with it. Thanks Allan.

**Comment by cgohlke** [comment]
Can this be backported to numpy 1.11.x? 
Scipy 0.18.0rc1 fails one test due to this:
```
Running unit tests for scipy
NumPy version 1.11.1
NumPy relaxed strides checking option: False
NumPy is installed in X:\Python35\lib\site-packages\numpy
SciPy version 0.18.0rc1
SciPy is installed in X:\Python35\lib\site-packages\scipy
Python version 3.5.2 (v3.5.2:4def2a2901a5, Jun 25 2016, 22:18:55) [MSC v.1900 64 bit (AMD64)]
nose version 1.3.7
<snip>
======================================================================
ERROR: test_mstats_basic.TestKruskal.test_result_attributes
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python35\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "X:\Python35\lib\site-packages\scipy\stats\tests\test_mstats_basic.py", line 708, in test_result_attributes
    res = mstats.kruskal(x, y)
  File "X:\Python35\lib\site-packages\scipy\stats\mstats_basic.py", line 1049, in kruskal
    H = 12./(ntot*(ntot+1)) * (sumrk**2/ngrp).sum() - 3*(ntot+1)
  File "X:\Python35\lib\site-packages\numpy\ma\core.py", line 3971, in __truediv__
    return true_divide(self, other)
  File "X:\Python35\lib\site-packages\numpy\ma\core.py", line 1118, in __call__
    result = self.f(da, db, *args, **kwargs)
ValueError: operands could not be broadcast together with shapes (2,) (2,5)

----------------------------------------------------------------------
Ran 23043 tests in 783.159s

FAILED (KNOWNFAIL=60, SKIP=1777, errors=1)
```

**Comment by charris** [comment]
@cgohlke Done. I wasn't planning on another release of 1.11.x, but could do. Currently there are also some enhancements in that branch as well as this bug fix so it will need some testing time before a release. The release schedule needs some discussion.

**Comment by cgohlke** [comment]
Thank you.

**Pull request merged**